library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java

```diff
@@ -347,6 +351,19 @@ public boolean isResumable() {
         return alwaysResume || (eTag != null && isNoIntegrity());
     }
 
+    public boolean shouldAllowTarUpdate(String mimeType) {
+        if (!allowTarUpdates) {
+            return false;
+        }
+
+        if (TAR_MIME_TYPE.equals(mimeType)) {
+            return true;
+        }
+
+        Log.e("Flag allowTarUpdates set but file not matching Tar mimeType, functionality will be disabled.");
```

Comment:
I would not abort the download just warn the user that it will not pause/truncate. I would keep a log here.